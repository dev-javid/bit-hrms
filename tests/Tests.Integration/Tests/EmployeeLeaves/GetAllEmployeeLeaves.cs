namespace Tests.Integration.Tests.EmployeeLeaves
{
    public class GetAllEmployeeLeaves(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "api/employee-leaves?getAll=True";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/EmployeeLeaves/GetAllEmployeeLeaves.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
