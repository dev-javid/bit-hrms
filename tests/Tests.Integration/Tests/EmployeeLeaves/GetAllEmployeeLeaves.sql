
INSERT INTO public.employee_leaves (id, company_id, employee_id, from_date, to_date, status, remarks,  created_by, created_on, modified_by, modified_on)
VALUES
(1, 998, 1, '2024-05-03', '2024-05-05', 0, 'Approved', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(2, 999, 1, '2024-01-03', '2024-01-05', 0, 'Approved', 0, '2024-05-04 03:33:32.227954-07', NULL, NULL),
(3, 999, 1, '2024-02-03', '2024-02-05', 0, 'Approved', 0, '2024-05-05 03:32:32.227954-07', NULL, NULL),
(4, 999, 1, '2024-03-03', '2024-03-05', 0, 'Approved', 0, '2024-05-03 03:37:32.227954-07', NULL, NULL),
(5, 999, 1, '2024-04-03', '2024-04-05', 0, 'Approved', 0, '2024-05-03 03:36:32.227954-07', NULL, NULL),
(6, 999, 1, '2024-05-03', '2024-05-05', 0, 'pending',  0, '2024-05-03 03:35:32.227954-07', NULL, NULL),
(7, 999, 1, '2024-06-03', '2024-06-05', 0, 'Pending',  0, '2024-05-03 03:34:32.227954-07', NULL, NULL);


INSERT INTO public.leave_policies(id, company_id, casual_leaves, earned_leaves_per_month, holidays, created_by, created_on,  modified_by, modified_on)
VALUES
(1, 998, 1, 2, 1, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(2, 999, 3, 4, 5, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL);