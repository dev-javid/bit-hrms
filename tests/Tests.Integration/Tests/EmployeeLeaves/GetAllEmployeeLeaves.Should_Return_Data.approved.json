{
    "page": 1,
    "limit": 10,
    "total": 6,
    "items": [
        {
            "employeeLeaveId": 3,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-02-05",
            "from": "2024-02-03",
            "createdOn": "2024-05-05T10:32:32.227954Z",
            "status": "Approved",
            "remarks": "Approved"
        },
        {
            "employeeLeaveId": 2,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-01-05",
            "from": "2024-01-03",
            "createdOn": "2024-05-04T10:33:32.227954Z",
            "status": "Approved",
            "remarks": "Approved"
        },
        {
            "employeeLeaveId": 4,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-03-05",
            "from": "2024-03-03",
            "createdOn": "2024-05-03T10:37:32.227954Z",
            "status": "Approved",
            "remarks": "Approved"
        },
        {
            "employeeLeaveId": 5,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-04-05",
            "from": "2024-04-03",
            "createdOn": "2024-05-03T10:36:32.227954Z",
            "status": "Approved",
            "remarks": "Approved"
        },
        {
            "employeeLeaveId": 6,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-05-05",
            "from": "2024-05-03",
            "createdOn": "2024-05-03T10:35:32.227954Z",
            "status": "Approved",
            "remarks": "pending"
        },
        {
            "employeeLeaveId": 7,
            "employeeId": 1,
            "employeeName": "Test-First-1 Test-Last-1",
            "to": "2024-06-05",
            "from": "2024-06-03",
            "createdOn": "2024-05-03T10:34:32.227954Z",
            "status": "Approved",
            "remarks": "Pending"
        }
    ]
}